source/simulator/assembly.cc

```diff
@@ -822,6 +822,21 @@ namespace aspect
 
     material_model->evaluate(scratch.material_model_inputs,
                              scratch.material_model_outputs);
+
+#ifdef DEBUG
+    {
+      // Check that the user filled all entries inside the material model and
+      // hasn't messed with the size of the objects:
+      static bool first = true;
+      if (first)
```

Comment:
Maybe add a comment to this to the effect that this really only checks the first time a matrix is assembled, but that that already catches the majority of cases?